src/lib.rs

```diff
@@ -549,7 +549,7 @@ impl<T: FftNum> RealToComplex<T> for RealToComplexEven<T> {
 
         let fftlen = self.length / 2;
         let buf_in = unsafe {
-            let ptr = input.as_mut_ptr() as *mut Complex<T>;
+            let ptr = input.as_ptr() as *mut Complex<T>;
```

Comment:
Is this really appropriate? RustFFT needs a mutable input vector since it uses it as a scratch space. Here `input` starts off as immutable, and then becomes mutable by casting the immutable pointer to a mutable one.

Reply:
Ah, you're correct, my mistake.